nearquake/utils/__init__.py

```
@@ -231,6 +231,38 @@ def generate_date_range(start_date, end_date, interval) -> tuple:
+
+    try:
+        start = datetime.strptime(start_date, "%Y-%m-%d").date()
+        end = datetime.strptime(end_date, "%Y-%m-%d").date()
+
+        if start >= end:
+            raise ValueError("start_date must be before end_date")
+        if interval <= 0:
+            raise ValueError("interval must be positive")
+
+    except ValueError:
+        raise ValueError("Invalid date format. Use YYYY-MM-DD.")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)